quickwit/quickwit-actors/src/mailbox.rs

```diff
@@ -340,13 +340,19 @@ impl<A: Actor> Inbox<A> {
         self.rx.try_recv()
     }
 
-    pub async fn recv_typed_message<M: 'static>(&self) -> Option<M> {
-        while let Ok(mut envelope) = self.rx.recv().await {
-            if let Some(msg) = envelope.message_typed() {
-                return Some(msg);
+    pub async fn recv_typed_message<M: 'static>(&self) -> Result<M, RecvError> {
```

Comment:
This is only used for tests, right?

Reply:
yes I can guard it with feature=testsuite I guess.